IntelliTect.TestTools.TestFramework/IntelliTect.TestTools.TestFramework/TestBuilder.cs

```diff
@@ -222,7 +182,150 @@ private string GetObjectDataAsJsonString(object obj)
             }
         }
 
+        private object GetTestBlock(IServiceScope scope, Type tbType, ILogger logger)
+        {
+            logger.Info(TestCaseName, tbType.ToString(), "Starting test block.");
+
+            try
+            {
+                return scope.ServiceProvider.GetRequiredService(tbType);
+            }
+            catch (InvalidOperationException ex)
+            {
+                // Probably worth moving these logs outside of the foreach so we don't have to duplicate the message
+                logger.Error(TestCaseName, tbType.ToString(), "Unable to find the test block instance OR all dependencies necessary.");
+                TestBlockException = ex;
+                return null;
+            }
+
+        }
+
+        private void SetTestBlockProperties(IServiceScope scope, ref object testBlockInstance, ILogger logger)
+        {
+            // Populate and log all of our properties
+            var properties = testBlockInstance.GetType().GetProperties();
+            foreach (var prop in properties)
+            {
+                if (!prop.CanWrite)
+                {
+                    logger.Debug(TestCaseName, testBlockInstance.GetType().ToString(), $"Unable to set property {prop}. No setter found.");
```

Comment:
A minor thing here, but a better message might be "Skipping property ..." rather than "Unable to set property..."

Reply:
Done with next commit